packages/storefront-lwc/src/modules/commerce/headerBasket/headerBasket.js

```diff
@@ -4,15 +4,31 @@
     SPDX-License-Identifier: BSD-3-Clause
     For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
 */
-import { LightningElement } from 'lwc';
+import { LightningElement, wire } from 'lwc';
 import { ShoppingBasket } from 'commerce/data';
+import { useQuery } from '@lwce/apollo-client';
+import gql from 'graphql-tag';
+
+const QUERY = gql`
+    query {
+        getBasketProductCount
+    }
+`;
 
 /**
  * Header basket component that should show up in the header
  */
 export default class HeaderBasket extends LightningElement {
     quantity = 0;
 
+    @wire(useQuery, {
+        query: QUERY,
+        lazy: false,
+    })
+    updateProductsTotal(response) {
+        this.quantity = response.data ? response.data.getBasketProductCount : 0;
```

Comment:
We should use the "initialized" property instead of checking for response.data. Alternatively, we can get rid of the function and just have a variable instead and use that variable in the component for loading, initialized, etc. Check out for ideas: https://blog.riand.com/2020/02/lwc-accessing-graphql-data-with-apollo.html

Reply:
Done